p2p/server/server_test.go

```diff
@@ -44,7 +44,30 @@ func wrapHost(tb testing.TB, h host.Host) Host {
 func TestServer(t *testing.T) {
 	const limit = 1024
 
-	mesh, err := mocknet.FullMeshConnected(5)
+	// Don't establish peer connections immediately.
+	// If we make the nodes connect to each other right away, there
+	// may be a problem with libp2p identify service. Namely, when a
+	// connection to a peer is established, identify request is sent
+	// to that peer, to which the peer sends identify response. Also,
+	// when a new handler is added to the host via SetStreamHandler,
+	// as it's done in p2p/server.Server, the identify response
+	// message is pushed to the peers currently connected to this
+	// node.
+	// When a server is created immediately following a connection
+	// from a peer, that peer may receive 2 identify response messages
+	// in quick succession: one which is indeed the response to
+	// initial identify request, and one which is push message
+	// generated by creation of the server. In some cases, the initial
+	// response may not contain the protocol of the freshly added
+	// server (hashProtocol="hs/1" in this case).
+	// The identify response message sent when SetStreamHandler is
+	// called will always contain the protocol being added, but due to
+	// how libp2p works, the responses may be processed in reverse
+	// order, and the set of protocols from the initial message will
+	// override the set of protocols in the second mesage (from
+	// SetStreamHandler). In this case, the peer will have a wrong
+	// idea about this node's protocols, and the test may fail.
+	mesh, err := mocknet.FullMeshLinked(5)
```

Comment:
How about putting these comments as a doc on the `FullMeshLinked` function?

Reply:
`FullMeshLinked` is a part of `libp2p`